packages/block-library/src/heading/autogenerate-anchors.js

```diff
@@ -0,0 +1,219 @@
+/**
+ * External dependencies
+ */
+import { isEmpty, isNil, deburr, trim } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { dispatch, select, subscribe } from '@wordpress/data';
```

Comment:
Ideally we should avoid these three global APIs (they are mostly there for third-parties and backward compatibility) but they can break in case of nested block editors.

Reply:
@youknowriad How they can break? Should we find another way for this PR here as well: https://github.com/WordPress/gutenberg/pull/30821/files#diff-76996844be36e9f96e35b67c600d137cfdf5011fb5b80b29ee28fa6becc5e7c7R7